cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverPluginTest.java

```diff
@@ -177,10 +172,15 @@ public void testPlugin() throws Exception {
 
       LocationFactory locationFactory = injector.getInstance(LocationFactory.class);
       String logBaseDir = cConf.get(LoggingConfiguration.LOG_BASE_DIR);
-      Location ns1LogBaseDir = locationFactory.create(namespaceDir).append("NS_1").append(logBaseDir);
 
-      FileMetaDataManager fileMetaDataManager = injector.getInstance(FileMetaDataManager.class);
-      waitTillLogSaverDone(fileMetaDataManager, loggingContext, "Test log message 59 arg1 arg2");
+      FileMetadataReader fileMetadataReader =
+        new FileMetadataReader(injector.getInstance(DatasetFramework.class),
+                               injector.getInstance(TransactionSystemClient.class),
+                               injector.getInstance(RootLocationFactory.class),
+                               injector.getInstance(Impersonator.class),
+                               "log.meta");
+
+      waitTillLogSaverDone(fileMetadataReader, loggingContext, "Test log message 59 arg1 arg2");
```

Comment:
good point, i made changes to that and also LogSaverPluginTest and removed the listFiles method in `FileMetadataManager`. so we only use `FileMetadataReader` for listing files.